libs/profile/src/lib/components/profile/profile.component.ts

```diff
@@ -25,13 +28,43 @@ export class ProfileComponent {
         });
     }
 
-    public onChangePassword(): void {
-        const ngPopoverRef = this.openPopup<void, { newPassword: string }>(this.changePasswordTemplate);
-        ngPopoverRef.afterClosed$.subscribe((result: NgPopoverCloseEvent<{ newPassword: string }>) => {
-            if (result.data.newPassword) {
-                this.authService.changePassword(result.data.newPassword);
-            }
-        });
+    public async onChangePassword(): Promise<void> {
+        if (await this.authService.isGmail()) {
+            this.openPopup<void, { any: any }>(this.changePasswordTemplateGmail);
+        } else {
+            const ngPopoverRef = this.openPopup<void, { newPassword: string; oldPassword: string }>(
+                this.changePasswordTemplate
+            );
+            ngPopoverRef.afterClosed$.subscribe(
+                (result: NgPopoverCloseEvent<{ newPassword: string; oldPassword: string }>) => {
+                    if (result.data.newPassword && result.data.oldPassword) {
+                        this.authService.changePassword(result.data.newPassword, result.data.oldPassword);
+                    }
+                }
+            );
+        }
+    }
+
+    public async onLoginWithGoogleForDelete(): Promise<void> {
+        await this.authService.deleteProfile('gmail');
+    }
+
+    public async onDeleteAccount(): Promise<void> {
+        if (await this.authService.isGmail()) {
+            const ngPopoverRef = this.openPopup<void, { confPassword: string }>(this.deleteAccountTemplateGmail);
+            ngPopoverRef.afterClosed$.subscribe((result: NgPopoverCloseEvent<{ confPassword: string }>) => {
+                if (result.data.confPassword != null) {
+                    this.authService.deleteProfile(result.data.confPassword);
+                }
+            });
+        } else {
+            const ngPopoverRef = this.openPopup<void, { confPassword: string }>(this.deleteAccountTemplate);
+            ngPopoverRef.afterClosed$.subscribe((result: NgPopoverCloseEvent<{ confPassword: string }>) => {
+                if (result.data.confPassword != null) {
```

Comment:
Meistens will man bei Javascript/Typescript Vergleiche mit === bzw. !== ausführen (siehe https://stackoverflow.com/questions/523643/difference-between-and-in-javascript)

Unabhängig davon schlägt die Logik hier fehl, wenn man ein leeres Passwort eingibt. Heißt der Fall des leeren Strings muss auch noch überprüft werden